woocommerce/woocommerce
# [Product Editor] Fix blank editor flash when loading product

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR gets rid of the blank flash of white between the loading state and the loaded editor.

It moves the handling of the loading state into the editor components themselves, so that the editor components can decide for themselves when to show the loading state.

It also adds an `EditorLoadingContext` that can be used by plugins to determine if the editor is loading.



### How to test the changes in this Pull Request:



With a WooCommerce dev env with the new product editor enabled (**WooCommerce** > **Settings** > **Advanced** > **Features** > **Experimental features**)...

1. Go to **Products** > **Add New**
2. Verify there is no blank flash of white when loading the editor... it should go from the loading state directly to the form
3. Add some variations on the **Variations** tab.
4. Click **Edit** on a variation in the list and verify that the editor loads the variation correctly.
5. Verify there is no blank flash of white when loading the editor... it should go from the loading state directly to the form
6. Install the following test plugin, which will be used to verify that plugins can use the editor loading state:
    - [test-43840-editor-loading.zip](https://github.com/woocommerce/woocommerce/files/14279081/test-43840-editor-loading.zip)
7. Verify the following footers added by the test plugin show properly:
    - "I'm always visible!" (red background) - always visible
    - "I'm only visible while the editor is loading!" (yellow background) - only visible while the editor is loading
    - "I'm only visible after the editor has loaded!" (black background) - only visible after the editor has loaded
8. Bonus points... Delete the following options using WCA Test Helper (or other means):
    - `woocommerce_block_product_tour_shown`
    - `woocommerce_product_editor_show_feedback_bar`
9. Make sure **WooCommerce** > **Settings** > **Advanced** > **Usage Tracking** > **Enable tracking** > **Allow usage of WooCommerce to be tracked** is enabled
10. Go to **Products** > **Add New**
11. Verify the welcome tour appears after the editor has loaded
12. Publish the product
13. Verify the feedback footer appears



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
